fe/fe-core/src/main/java/com/starrocks/sql/ast/CTERelation.java

```diff
@@ -17,6 +17,13 @@ public CTERelation(int cteId, String name, List<String> columnOutputNames, Query
         this.cteQuery = cteQuery;
     }
 
+    public CTERelation(String name, List<String> columnOutputNames, QueryRelation cteQuery) {
+        this.cteId = -1;
```

Comment:
when you update the cteId after ? I think  maybe we need a IdGenerator here.

Reply:
done